addons/docs/jest-transform.js

```diff
@@ -0,0 +1,20 @@
+const mdx = require('@mdx-js/mdx');
+const babel = require('babel-jest');
+const deasyncPromise = require('deasync-promise');
+const createCompiler = require('./mdx-compiler-plugin');
+
+const compilers = [createCompiler({})];
+
+module.exports = {
+  process(src, filename, config, options) {
+    let result = deasyncPromise(mdx(src, { compilers, filepath: filename }));
```

Comment:
I wasn't aware jest transformers had to be synchronous.
Just for future reference, here's why: https://github.com/facebook/jest/issues/5556

(this comment is mainly for future-me)